float32 sail_angle
float32 rudder_angle